File: Web/src/Components/Policy/PolicyClasses/Policy.js
import { makeObservable, observable, runInAction } from 'mobx';
import axios from 'axios';

export default class Policy {
    id = 0;
    name = '';
    rules = [];
    updateCount = 0;

    constructor({ id, policyName, rules, updateCount }) {
        makeObservable(this, {
            id: observable,
            name: observable,
            rules: observable,
            updateCount: observable,
        });
        this.id = id;
        this.name = policyName;
        this.rules = rules;
        this.updateCount = updateCount;
    }

    updatePolicy = ({ name, rules }) => {
        const dataToSend = {
            id: this.id,
            policyName: name,
            numberOfRules: rules.length,
            rules: rules.map((rule) => rule.id),
            updateCount: this.updateCount + 1,
        };

        return axios
            .put(`/policy/${this.id}`, dataToSend)
            .then((response) => {
                if (response.status === 200) {
                    runInAction(() => {
                        this.name = name;
                        this.rules = rules.map((rule) => rule.id);
                        this.updateCount += 1;
                    });
                    return true;
                }
                return false;
            })
            .catch((error) => {
                console.log(error);
                return false;
            });
    };
}

File: Server/Project/Boundaries/Policy.py
from Boundaries import Rule


class Policy:
    def __init__(self, policy_id=0, policy_name: str = "Policy", number_of_rules: int = 0,
                 rules=[], update=0):
        self.id = policy_id
        self.policyName = policy_name
        self.numberOfRules = number_of_rules
        self.rules = rules
        self.updateCount = update

    def __str__(self):
        return f"ID:{self.id}, " \
               f"PolicyName:{self.policyName}, " \
               f"NumberOfRules:{self.numberOfRules}, " \
               f"Rules:{self.rules}, " \
               f"Update:{self.updateCount}"
